ocaml/testsuite/tests/pattern-guards/nested_guards.ml

```diff
@@ -0,0 +1,121 @@
+(* TEST
```

Comment:
I appreciate that you've split up these tests, but I suspect that this change would have made more sense in a separate PR so the squashed-and-merged git history is more intelligible. (And to make review a bit easier.) Would this be easy for you to do?

Reply:
This is pretty easy, partially because the changes are mostly logically separate from this PR's, which is also a good reason to make a separate PR for them. Done!